File: calculator.py
import sys

def add(a, b):
    return a + b

def subtract(a, b):
    return a - b

def multiply(a, b):
    return a * b

def divide(a, b):
    if b == 0:
        print("Error: Division by zero is undefined.")
        return None
    return a / b

def power(a, b):
    return a ** b

def modulus(a, b):
    return a % b

def get_number(prompt):
    while True:
        try:
            return float(input(prompt))
        except ValueError:
            print("Invalid input. Please enter a numeric value.")

def main():
    print("Welcome to the Simple Calculator!")
    operations = {
        '1': ('Add', add),
        '2': ('Subtract', subtract),
        '3': ('Multiply', multiply),
        '4': ('Divide', divide),
        '5': ('Power', power),
        '6': ('Modulus', modulus),
        '7': ('Exit', None)
    }

    while True:
        print("\nSimple Calculator")
        for key, (name, _) in operations.items():
            print(f"{key}. {name}")

        choice = input("Select operation (1-7): ")

        if choice not in operations:
            print("Invalid selection. Please choose a valid option.")
            continue

        if choice == '7':  # Exit
            print("Exiting calculator. Goodbye!")
            sys.exit()

        op_name, op_func = operations[choice]
        num1 = get_number("Enter first number: ")
        num2 = get_number("Enter second number: ")

        result = op_func(num1, num2)
        if result is not None:
            print(f"Result of {op_name.lower()}ing {num1} and {num2} = {result}")

if __name__ == "__main__":
    main()

File: calculator.js
// Basic Calculator in JavaScript
function add(a, b) {
    return a + b;
}

function subtract(a, b) {
    return a - b;
}

function multiply(a, b) {
    return a * b;
}

function divide(a, b) {
    if (b === 0) {
        throw new Error('Cannot divide by zero');
    }
    return a / b;
}

// Example usage:
console.log('Add: 2 + 3 =', add(2, 3));
console.log('Subtract: 5 - 2 =', subtract(5, 2));
console.log('Multiply: 4 * 3 =', multiply(4, 3));
console.log('Divide: 10 / 2 =', divide(10, 2));

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter the first number: ', (inputA) => {
    rl.question('Enter the second number: ', (inputB) => {
        const a = parseFloat(inputA);
        const b = parseFloat(inputB);
        if (isNaN(a) || isNaN(b)) {
            console.log('Invalid input. Please enter valid numbers.');
        } else {
            console.log('Add:', a, '+', b, '=', add(a, b));
            console.log('Subtract:', a, '-', b, '=', subtract(a, b));
            console.log('Multiply:', a, '*', b, '=', multiply(a, b));
            try {
                console.log('Divide:', a, '/', b, '=', divide(a, b));
            } catch (err) {
                console.log('Divide:', err.message);
            }
        }
        rl.close();
    });
});
